File: src/01-easy/ascii-art/solution.py
L=int(input());H=int(input());T=input().upper()
rows=[input()for _ in range(H)]
for i in range(H):
    line=""
    for c in T:
        idx=ord(c)-65
        if idx<0 or idx>25:idx=26
        line+=rows[i][idx*L:idx*L+L]
    print(line)

File: src/01-easy/ascii-art/solution.clj
(ns Solution (:gen-class)
  (:require [clojure.string :as str]))
(defn -main [& args]
  (let [L (Integer/parseInt (read-line))
        H (Integer/parseInt (read-line))
        T (str/upper-case (read-line))
        rows (vec (repeatedly H read-line))]
    (doseq [i (range H)]
      (println
        (apply str
          (map (fn [c]
                 (let [idx (- (int c) 65)
                       idx (if (or (< idx 0) (> idx 25)) 26 idx)]
                   (subs (rows i) (* idx L) (+ (* idx L) L))))
               T))))))
